File: update_football_images_wiki.py
import json
import re
import urllib.request
import urllib.parse
import time

def get_wiki_image(title):
    try:
        url = f"https://en.wikipedia.org/w/api.php?action=query&titles={urllib.parse.quote(title)}&prop=pageimages&format=json&pithumbsize=400"
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode())
            pages = data['query']['pages']
            for page_id in pages:
                if 'thumbnail' in pages[page_id]:
                    return pages[page_id]['thumbnail']['source']
    except Exception as e:
        print(f"Failed to fetch {title}: {e}")
    return "football.jpg"

mappings = {
    "Match Ball": "ball_gen.png",
    "Gloves": "gloves_gen.png",
    "Jersey": "authentic_jersey.jpg",
    "Cleats": get_wiki_image("Cleat_(shoe)"),
    "Shoulder Pads": get_wiki_image("Shoulder_pads_(American_football)"),
    "Rib Protector": get_wiki_image("Flak_jacket"),
    "Kicking Tee": get_wiki_image("Tee_(sports)"),
    "Helmet": get_wiki_image("Football_helmet"),
    "Knee Pads": get_wiki_image("Knee_pad"),
    "Training Pants": get_wiki_image("Sweatpants")
}

with open('products.js', 'r', encoding='utf-8') as f:
    content = f.read()

json_str = content[content.find('['):content.rfind(']')+1]
products = json.loads(json_str)

updated_count = 0
for p in products:
    if p['category'] == 'Football':
        for key, img in mappings.items():
            if key.lower() in p['name'].lower():
                p['image'] = img
                updated_count += 1
                break

new_content = "const products = " + json.dumps(products, indent=4) + ";\n"
with open('products.js', 'w', encoding='utf-8') as f:
    f.write(new_content)

print(f"Successfully updated {updated_count} football products with accurate images.")

File: update_football_images_wiki.php
<?php
function get_wiki_image($title) {
    $url = "https://en.wikipedia.org/w/api.php?action=query&titles=" . urlencode($title) . "&prop=pageimages&format=json&pithumbsize=400";
    $options = [
        "http" => [
            "header" => "User-Agent: Mozilla/5.0\r\n"
        ]
    ];
    $context = stream_context_create($options);
    $response = file_get_contents($url, false, $context);
    if ($response) {
        $data = json_decode($response, true);
        if (isset($data['query']['pages'])) {
            $pages = $data['query']['pages'];
            foreach ($pages as $page) {
                if (isset($page['thumbnail']['source'])) {
                    return $page['thumbnail']['source'];
                }
            }
        }
    }
    return "football.jpg";
}

$mappings = [
    "Match Ball" => "ball_gen.png",
    "Gloves" => "gloves_gen.png",
    "Jersey" => "authentic_jersey.jpg",
    "Cleats" => get_wiki_image("Cleat_(shoe)"),
    "Shoulder Pads" => get_wiki_image("Shoulder_pads_(American_football)"),
    "Rib Protector" => get_wiki_image("Flak_jacket"),
    "Kicking Tee" => get_wiki_image("Tee_(sports)"),
    "Helmet" => get_wiki_image("Football_helmet"),
    "Knee Pads" => get_wiki_image("Knee_pad"),
    "Training Pants" => get_wiki_image("Sweatpants")
];

$content = file_get_contents('products.js');
$start = strpos($content, '[');
$end = strrpos($content, ']');
$json_str = substr($content, $start, $end - $start + 1);

$products = json_decode($json_str, true);
if (!$products) {
    die("Error parsing JSON. Ensure products.js contains valid JSON inside the array assignment.");
}

$updated_count = 0;
foreach ($products as &$p) {
    if ($p['category'] === 'Football') {
        foreach ($mappings as $key => $img) {
            if (stripos($p['name'], $key) !== false) {
                $p['image'] = $img;
                $updated_count++;
                break;
            }
        }
    }
}

$new_content = "const products = " . json_encode($products, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) . ";\n";
file_put_contents('products.js', $new_content);

echo "Successfully updated " . $updated_count . " football products with accurate wiki/local images.\n";
?>
